catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/QueryResponseImpl.java

```diff
@@ -302,7 +302,7 @@ private Result handleTake() {
         Result result = null;
         try {
             result = queue.take();
-            if (result == poisonPillResult) {
+            if (result == POISON_PILL_RESULT) {
```

Comment:
@stustison maybe I'm the crazy one but I think it should be okay. I believe that class instances containing or referencing static members (even in a collection) will still be gc'd if the reference to the class instance itself is no longer referenced. haha or I may be completely misunderstanding your comment :-) In any case a profiler can be used for verification.